### PR TITLE
AB#76206: Home controller

### PR DESCRIPTION
**Pull request by MoizIqbal97**
Adds ConfigKey string varliables
Contact Controller moved out to its own Controller
Home Controller added
Config Service updated to use IMemoryCache
_FeedbackMessage view ported

### CODE REVIEW

File: src/Submissions/Api/Controllers/HomeController.cs [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+                NetworkRegistration = await _configService.GetSiteConfigValue(ConfigKey.HomepageNetworkRegistration, true, ""),
+                NetworkRegistration2 = await _configService.GetSiteConfigValue(ConfigKey.HomepageNetworkRegistration2, true, ""),
+                RequireSamplesCollected = await _configService.GetSiteConfigValue(ConfigKey.HomepageSearchRadioSamplesCollected, true, ""),
+                AccessExistingSamples = await _configService.GetSiteConfigValue(ConfigKey.HomepageSearchRadioAccessSamples, true, ""),
+                FinalParagraph = await _configService.GetSiteConfigValue(ConfigKey.HomepageFinalParagraph ,true,""),
+                ResourceRegistrationButton = await _configService.GetSiteConfigValue(ConfigKey.RegisterBiobankTitle, true, ""),
+                NetworkRegistrationButton = await _configService.GetSiteConfigValue(ConfigKey.RegisterNetworkTitle, true, "")
+            });
+        }
+      //  public ActionResult Cookies() => View(); TODO copy views over
```
Comment (AndyRae) [inline src/Submissions/Api/Controllers/HomeController.cs:45]: If we're not adding the home / contact views at this point, it's worth adding a PBI to make sure we do pick them up.

**Review by AndyRae — APPROVED** [review]

File: src/Submissions/Api/Controllers/ContactController.cs [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+
+namespace Biobanks.Submissions.Api.Controllers
+{
+    public class ContactController : Controller
+    {
+        private readonly INetworkService _networkService;
+        private readonly IOrganisationDirectoryService _organisationService;
+        private readonly IMapper _mapper;
+        private readonly IMemoryCache _memoryCache;
+        //private readonly IEmailService _emailService; //TODO: Email Service has not be ported yet
```
Comment (dcl10) [inline src/Submissions/Api/Controllers/ContactController.cs:24]: ⛏️ If you're not using the commented code in this file anymore, just delete it

File: src/Submissions/Api/Controllers/HomeController.cs [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+    {
+        private readonly IConfigService _configService;
+
+        public HomeController(IConfigService configService)
+        {
+            _configService = configService;
+        }
+
+        // GET: Home
+        public async Task<IActionResult> IndexAsync()
```
Comment (dcl10) [inline src/Submissions/Api/Controllers/HomeController.cs:24]: Does this not need a `[HttpGet]` decorator?

File: src/Submissions/Api/Controllers/HomeController.cs (hunk 1 above)
Comment (dcl10) [inline src/Submissions/Api/Controllers/HomeController.cs:45]: ⛏️ remove if not needed

**Review by dcl10 — CHANGES_REQUESTED** [review]

File: src/Submissions/Api/Controllers/HomeController.cs (hunk 3 above)
Comment (MoizIqbal97) [reply src/Submissions/Api/Controllers/HomeController.cs:24]: We are inheriting from the `Controller` class, the convention for this is `/<ControllerName>/<ActionMethodName>` .
Index method is a special action name where it is served from just `/<ControllerName>/` 

The decorates are only used for API controllers which uses the `ControllerBase` class and the `[ApiController]` attribute

(answer plagarised from @beforan XD )

**Review by dcl10 — APPROVED** [review]

**Review by AndyRae — APPROVED** [review]

**Pull request merged**